sgt-back-end/index.js

```diff
@@ -0,0 +1,140 @@
+const express = require('express');
+const app = express();
+const pg = require('pg');
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+  select *
+    from "grades"`;
+    db.query(sql)
+    .then((result => {
+      res.status(200).json(result.rows);
+    }))
+    .catch(err => {
+      res.status(500).json({
+        error: 'Something went wrong'
+      })
+    })
+})
+
+app.use(express.json())
+
+app.post('/api/grades', (req, res) => {
+  if(req.body.name === undefined || req.body.course === undefined || req.body.score === undefined
+    || req.body.score < 1 || req.body.score > 100 || !Number.isInteger(parseInt(req.body.score))) {
```

Comment:
Since `parseInt` forces the `score` to be an integer if it's fractional (for example, `parseInt(13.54, 10)` returns `13`), so fractional `score`s will pass, even though they shouldn't.